### PR TITLE
Add ci jobs for lcg-release deployment

### PR DESCRIPTION
**Pull request by haralmha**

### CODE REVIEW

**Comment by haralmha** [comment]
* Do I need to set `K4_JOBTYPE`?
* Should I add a tag?
@vvolkl

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -83,6 +83,33 @@ build-spack-nightlies:
...
+    needs:
+        - "setup_spack_push"
+    only:
+      refs:
+          - schedules # Only execute this on scheduled "nightly" pipelines
+    script:
+        - source spack/share/spack/setup-env.sh
+        - git clone https://gitlab.cern.ch/sft/sft-spack-repo.git
+        - cd sft-spack-repo
+        - source environments/lcg-release/setup_clingo_centos7.sh
```
Comment (vvolkl) [inline .gitlab-ci.yml:98]:
```suggestion
```
The build machine is already setup with clingo. Not putting this here helps with using this CI job on other platforms

Comment (vvolkl) [reply .gitlab-ci.yml:98]: Yes, please add the `k4-build-spack-release` tag and make the job conditional on `$K4_JOBTYPE == "LCG_Release"` for example

**Comment by vvolkl** [comment]
For the deploy step, take https://github.com/key4hep/key4hep-spack/blob/release/.gitlab-ci.yml#L205 as template instead of the nightlies.

**Pull request merged**